system/threads.cpp

```diff
@@ -224,21 +233,25 @@ void startWorkerFunction(unsigned int fp, unsigned int args, unsigned int tls, i
 		utilityWorker->postMessage(threadingObject);
 		return;
 	}
-	struct ThreadSpawnInfo spawnInfo;
-	spawnInfo.func = fp;
-	spawnInfo.args = args;
-	spawnInfo.tls = tls;
-	spawnInfo.tid = newThreadId;
-	spawnInfo.stack = stack;
-	spawnInfo.ctid = ctid;
-
-	threadMessagingQueue.send(spawnInfo);
+	QueueMessage message;
+	message.type = QueueMessageType::SPAWN_THREAD;
+	message.spawnInfo.func = fp;
+	message.spawnInfo.args = args;
+	message.spawnInfo.tls = tls;
+	message.spawnInfo.tid = newThreadId;
+	message.spawnInfo.stack = stack;
+	message.spawnInfo.ctid = ctid;
+
+	threadMessagingQueue.send(message);
 }
 
 [[cheerp::genericjs]]
-void spawnThreadFromUtility()
+void waitForMessage();
+
+[[cheerp::genericjs]]
+void spawnThreadFromUtility(QueueMessage message)
```

Comment:
accept directly a `ThreadSpawnInfo` here. it's the only member of the union used in this function

Reply:
Relic from in-between states. Changed to only accept the ThreadSpawnInfo